ResearchKit/Common/ORKSurveyAnswerCellForText.m

```diff
@@ -327,12 +351,12 @@ - (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRang
 }
 
 - (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
-    return YES;
+    return [self isAnswerValidForTextField:textField];
```

Comment:
Looking back at other survey cells. The current user experience does not show a validity alert unless the user hits the Next button. So we should actually just revert this to `return YES`. The validation alert will show up when the user hits Next.